### ISSUE TITLE
why my 'workers=8,use_multiprocessing=True' do not work while training?

### ISSUE THREAD

**Issue by JWenBin** [tensorflow/tensorflow#47633]
**Here is my code：**
train_model_input=generate_arrays_from_dataframe( data.sample(frac=1) )
history = model.fit(train_model_input,  epochs=1, verbose=1, validation_split=0.0,steps_per_epoch= math.ceil( train_row_len/256),  workers=8,use_multiprocessing=True)  

I find it only 100% CPU-Util not the expect 800%.
![图片](https://user-images.githubusercontent.com/49393828/110285880-e4be8f00-801e-11eb-8e83-b26b4297729e.png)

And my GPU-Util is 0% although GPU memory is fully used
![图片](https://user-images.githubusercontent.com/49393828/110285828-ce183800-801e-11eb-9605-1e4289ae39b1.png)

How could I Increase the utilization of my CPU/GPU?
thank you!

**Comment by JWenBin** [comment]
should i read my data(pd.read_csv) inside generate_arrays_from_file(path) ?  But i find my generate_arrays_from_dataframe entered 8 times.

**Comment by JWenBin** [comment]
I tried keras.utils.Sequence too. The GPU-Util is between 0%-22%, and CPU-Util is about 100%. **How Can I Increase the utilization of my CPU to 800%?**

my DataGenerator code is:
class DataGenerator(keras.utils.Sequence):
    def __init__(self, df, batch_size=256,):
        self.batch_size = batch_size
        self.dataframe = df
        self.indexes = np.arange(len(self.df))
......

my training code is:
train_model_input = DataGenerator.DataGenerator(df.sample(frac=1)) 

# strategy = tf.distribute.MirroredStrategy()
# train_dist_dataset = strategy.experimental_distribute_dataset(train_model_input) # DataGenerator
# print ('Number of devices: {}'.format(strategy.num_replicas_in_sync))
# with strategy.scope():  
model = MIND(user_feature_columns,item_feature_columns,dynamic_k=False,p=1,k_max=1,num_sampled=5,user_dnn_hidden_units=(64, embedding_dim))
model.compile(optimizer="adam", loss=sampledsoftmaxloss)  # "binary_crossentropy")
history = model.fit(train_model_input, epochs=1, verbose=1, validation_split=0.0,steps_per_epoch=math.ceil( train_row_len/256), workers=8,use_multiprocessing=True) 
.......

Any tips will be great, thanks!

**Comment by nikitamaia** [comment]
Hi @JWenBin, to help with the debugging process please provide a colab notebook with reproducible code. Thanks!

**Comment by JWenBin** [comment]
> 
> 
> Hi @JWenBin, to help with the debugging process please provide a colab notebook with reproducible code. Thanks!

I will continue it one month later. Thanks!

**Comment by nikitamaia** [comment]
Closing this issue due to lack of activity. Please comment or reopen if you have reproducible code. Thanks!

**Comment by renguote** [comment]
Is this problem solved?

**Comment by shuDaoNan9** [comment]
https://github.com/tensorflow/tensorflow/issues/57903#issuecomment-1263162024
